### PR TITLE
rpcdaemon: add get_capacity on StreamWriter

### PR DESCRIPTION
**Pull request by lupin012**

### CODE REVIEW

File: silkworm/rpc/transport/stream_writer.hpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ class StreamWriter {
     virtual Task<void> open_stream() = 0;
     virtual Task<void> close_stream() = 0;
     virtual Task<std::size_t> write(std::string_view content, bool last) = 0;
+    virtual size_t get_capacity() = 0;
```
Comment (canepat) [inline silkworm/rpc/transport/stream_writer.hpp:34]: It should be declared `const noexcept`

File: silkworm/rpc/http/connection.hpp [hunk 2]
```diff
@@ -65,6 +65,7 @@ class Connection : public StreamWriter {
     /* StreamWriter Interface */
     Task<void> open_stream() override;
     Task<void> close_stream() override;
+    size_t get_capacity() override { return 4 * 1024; }
```
Comment (canepat) [inline silkworm/rpc/http/connection.hpp:68]: Better use a constant instead of an hard-coded magic number

File: silkworm/rpc/ws/connection.hpp [hunk 3]
```diff
@@ -57,6 +57,7 @@ class Connection : public StreamWriter {
     // Methods of StreamWriter interface
     Task<void> open_stream() override { co_return; }
     Task<void> close_stream() override { co_return; }
+    size_t get_capacity() override { return 5 * 1024 * 1024; }
```
Comment (canepat) [inline silkworm/rpc/ws/connection.hpp:60]: Better use a constant instead of an hard-coded magic number

File: silkworm/rpc/json/stream.hpp [hunk 4]
```diff
@@ -44,9 +44,7 @@ struct DataChunk {
 //! Stream can be used to send big JSON data split into multiple fragments.
 class Stream {
   public:
-    inline static constexpr std::size_t kDefaultCapacity{5 * 1024 * 1024};
-
-    Stream(boost::asio::any_io_executor& executor, StreamWriter& writer, std::size_t buffer_capacity = kDefaultCapacity);
+    Stream(boost::asio::any_io_executor& executor, StreamWriter& writer, size_t new_buffer_capacity = 0);
```
Comment (canepat) [inline silkworm/rpc/json/stream.hpp:47]: `new_buffer_capacity` -> `buffer_capacity`

Moreover, I suggest defaulting `buffer_capacity` value to `writer_.get_capacity()` directly here

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/rpc/transport/stream_writer.hpp (hunk 1 above)
Comment (lupin012) [reply silkworm/rpc/transport/stream_writer.hpp:34]: ok

File: silkworm/rpc/http/connection.hpp (hunk 2 above)
Comment (lupin012) [reply silkworm/rpc/http/connection.hpp:68]: ok

File: silkworm/rpc/ws/connection.hpp (hunk 3 above)
Comment (lupin012) [reply silkworm/rpc/ws/connection.hpp:60]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**